flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/reconciler/deployment/ApplicationReconcilerTest.java

```diff
@@ -706,4 +707,47 @@ public void testRestartUnhealthyEvent() throws Exception {
         reconciler.reconcile(deployment, context);
         Assertions.assertEquals(MSG_RESTART_UNHEALTHY, eventCollector.events.remove().getMessage());
     }
+
+    @Test
+    public void testReconcileIfUpgradeModeNotAvailable() throws Exception {
```

Comment:
Could you please add comments to the test @gyfora? It's really hard to follow what it actually does.

Reply:
done